com.civicrm.hrjobroles/js/src/job-roles/services/date-validation.js

```diff
@@ -0,0 +1,78 @@
+define([
+    'job-roles/services/services',
+    'common/moment'
+], function (module, moment) {
+    /**
+     * Service responsible for validating dates in HRJobRoles
+     * @constructor
+     */
+    module.service('DateValidation', ['HR_settings', function (HR_settings) {
+        var me = this;
+
+        me._error = function (error_msg, fields) {
+            throw new Error(error_msg, fields[0]);
+        };
+
+        me.defaultFormat = HR_settings.DATE_FORMAT;
+
+        /**
+         *
+         * @param {function} error
+         * @throws TypeError
+         */
+        me.setErrorCallback = function setErrorCallback(error) {
+            if (typeof error === 'function') {
+                me._error = error;
+            } else {
+                throw new TypeError('Error callback must be a function.');
+            }
+        };
+
+        me.dateFormats = [
+            'x',
+            'YYYY-MM-DD'
+        ];
+
+        me.defaultFormat && me.dateFormats.push(me.defaultFormat.toUpperCase());
+
+        me.validate = function validate(start, end) {
+
+            if (start instanceof Date) {
+                start = start.getTime();
+            }
+
+            if (end instanceof Date) {
+                end = end.getTime();
+            }
+
+            var start_date = moment(start, me.dateFormats, true);
+            var end_date = moment(end, me.dateFormats, true);
+
+            checkIfValuesAreValid(start_date, ['start_date']);
+
+            if (end !== '' && typeof end !== 'undefined' && end !== null) {
+                checkIfValuesAreValid(end_date, ['end_date']);
+            }
+
+            if (start && end_date.isValid()) {
```

Comment:
What's the logic in this `if`? You are checking if `start` (a timestamp) exists and that `end_date` (a moment object) is valid, why the inconsistency between the start and end date?

Also, you just checked in `checkIfValuesAreValid` if `end_date.isValid()`, why redoing it right after it?